components/camera/rtsp/decode.go

```diff
@@ -2,26 +2,146 @@ package rtsp
 
 import (
 	"bytes"
+	"context"
+	"fmt"
 	"image"
-	"image/jpeg"
+	"unsafe"
 
-	"github.com/aler9/gortsplib/v2/pkg/format"
-	"github.com/pion/rtp"
 	"github.com/pkg/errors"
+
+	"go.viam.com/rdk/gostream/ffmpeg"
+	"go.viam.com/rdk/rimage"
+	"go.viam.com/rdk/utils"
 )
 
-type decoder func(pkt *rtp.Packet) (image.Image, error)
+// Decoder the decoder created for the RTSP stream.
+type Decoder struct {
+	codec       ffmpeg.Codec
+	context     ffmpeg.CodecContext
+	fmtCxt      ffmpeg.FormatContext
+	videoStream int
+	cancelCtx   context.Context
+	cancelFunc  func()
+}
+
+// NewDecoder returns a new decoder.
+func NewDecoder(ctx context.Context, addr string) (*Decoder, error) {
+	dec := &Decoder{}
+	dec.fmtCxt = ffmpeg.AllocFormatContext()
+	dec.cancelCtx, dec.cancelFunc = context.WithCancel(context.Background())
+
+	dict, err := ffmpeg.NewDictionary(ctx, ffmpeg.DictionaryEntries{
+		// Use `ffmpeg -h full` for a complete list of options
+		"rtsp_transport": "tcp",
+		"scan_all_pmts":  "1",
+		"sdp_flags":      "rtcp_to_source",
```

Comment:
[info] Found section of [rtcp.c](https://github.com/FFmpeg/FFmpeg/blob/master/libavformat/rtsp.c#L1524) where ports are setup for both RTP and RTCP.

Did not realize that IP Cameras (RTSP Servers) commonly provide RTCP feedback support. But also want to make sure that it still makes sense to send RTCP data when using TCP transport for video stream.

Reply:
I orignally added this option because when I stopped streaming the RTSP server would close the session and I'd get EOF errors client-side (in the RDK). This was my attempt to send packets periodically to keep that session open. This may not be needed anymore though since we're always streaming from the source now.